tests/integration/test_charm.py

```diff
@@ -16,35 +14,66 @@
 METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
 APP_NAME = METADATA["name"]
 
+INI_PATH = "/etc/pgbouncer/pgbouncer.ini"
+USERLIST_PATH = "/etc/pgbouncer/userlist.txt"
+
 
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test: OpsTest):
-    """Build the charm-under-test and deploy it together with related charms.
+    """Build and deploy pgbouncer charm."""
+    charm = await ops_test.build_charm(".")
+    resources = {
+        "pgbouncer-image": METADATA["resources"]["pgbouncer-image"]["upstream-source"],
+    }
+    await ops_test.model.deploy(
+        charm,
+        resources=resources,
+        application_name=APP_NAME,
+        config={"pgb_admin_users": "juju-admin"},
+    )
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
 
-    Assert on the unit status before any relations/configurations take place.
+async def test_user_management(ops_test: OpsTest):
+    """Test user management through config and actions.
+
+    We complete the following steps, verifying each action has the expected output:
+    - Create a user through `pgb_admin_users` config variable
+    - Change this user's password
+    - Add a new user through the `add-user` action
+    - Check we have all three expected users, using the `get-users` action
+    - Remove the created users in preparation for the next test, and check that they have been
+      removed using the `get-users` action
+
+    TODO examine userlist.txt directly to ensure changes are implemented as expected.
```

Comment:
You might want to do this before merging so that you can ensure all features work

Reply:
This has been added :)